### ISSUE TITLE
Excluded User Agent still tracked

### ISSUE THREAD

**Issue by dandv** [piwik/piwik#7978]
I had configured the following string under the Websites -> Global list of user agents to exclude:

> "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20100101 Firefox/10.0"

![image](https://cloud.githubusercontent.com/assets/33569/7783213/8bf38aac-00ee-11e5-8f86-15e0bb6d9299.png)

It belongs to the Monitor.us service.

Yet later I still saw that bot in the Real-time Visitors:

![image](https://cloud.githubusercontent.com/assets/33569/7783219/e2edcbce-00ee-11e5-9268-fbb661d7bf3e.png)

Do I maybe need to add the user agent string without quotes? By the way, there's no explanation of the format in which UA strings should be entered in that box. Presumably, one per line, but who knows? CSV format? Quoted or not?

**Comment by sgiehl** [comment]
Looking at the code it seems to be a unquoted comma separated list. But that might not be that good at all, as some useragents might also contain a comma.
Nevertheless it should be explained which format is used.

**Comment by mattab** [comment]
> Do I maybe need to add the user agent string without quotes?

Yes you need to set the user agent without quote. Does it work for you after removing the double quotes?

**Comment by dandv** [comment]
IMO it would be most intuitive and easiest to paste user agents one per line, and have Piwik automatically take care of enclosing double quotes. These are the two ways users will find the UA strings, from various sites (without quotes), or from server logs (with quotes).

Comma-separate adds unnecessary complexity and potential for breakage if comas aren't escaped correctly.

**Comment by mattab** [comment]
Actually the textarea field should accept one user agent per line, not comma separated. If one user agent per line does not work, please let us know, it's a bug we would fix ASAP

**Comment by dandv** [comment]
@mattab, see @sgiehl's comment above.

**Comment by saleemkce** [comment]
* current good commit https://github.com/piwik/piwik/commit/b8cafeee25b6422cdf829441a63eba0e778cf529 .Previous commit references/branches that you see here were deleted because they included some worng merge-commits in my local master.

> IMO it would be most intuitive and easiest to paste user agents one per line, and have Piwik automatically take care of enclosing double quotes.

* This has been fixed. Fix will take care of quotes if present

> Actually the textarea field should accept one user agent per line, not comma separated. If one user agent per line does not work, please let us know, it's a bug we would fix ASAP

* I tested this scenario extensively with 4 different browsers (IE, Chrome, FF & Opera)/Windows OS today. I didn't face any incorrect tracking. But when quotes were present before the commit I made, it failed to work correctly.

**Comment by mattab** [comment]
Suggested fix:
 * when user inputs a user agent with double quotes, we remove the enclosing double quotes (+ trim) before testing against the visitor user agent

**Comment by saleemkce** [comment]
@mattab The fix I posted handles the "remove the enclosing double quotes" part.  And the function **checkAndReturnCommaSeparatedStringList** already does the "(+ trim)" part. So, could you suggest what else we should consider or is the current fix sufficient?

**Comment by sgiehl** [comment]
as mentioned in #8050 useragents might contain quotes, so simply removing them is not accurate

**Comment by mattab** [comment]
we should remove the enclosing quotes only, eg. using the second parameter of PHP trim function

**Comment by mnapoli** [comment]
Currently WIP in #8448

**Comment by tsteur** [comment]
There is a PR for that but it's targeting 3.0 . Can we maybe move this issue out of 2.15 into 3.0? Maybe a solution that is ok for now would be to just mention in the inline help to not put them into quotes?

**Comment by mattab** [comment]
I'm not sure what's happened with this issue, 
but the bug is still active.

Reproduce:

* Go to Administration > Websites > Edit
* Under "Excluded User Agents", enter `iplabel` 
* Save
* Visit `matomo.php?idsite=1&rec=1&test&debug=1&ua=Mozilla/5.0 (X11; CrOS x86_64 8172.45.0 ;iplabel) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.64 Safari/537.36&debug=1` to generate a request with a user agent containing `iplabel`
* Got: the visit is tracked. Expected instead: visit excluded.
* Note: entering `iplabel` in Administration > Global websites settings > Global list of user agents to exclude, works as expected.
